test/appender/test_appender.cpp

```diff
@@ -416,3 +416,25 @@ TEST_CASE("Test alter table in the middle of append", "[appender]") {
 		REQUIRE_THROWS(appender.Close());
 	}
 }
+
+TEST_CASE("Test appending to a different database file", "[appender]") {
```

Comment:
Thanks for the PR!

Can we test with:
- Wrong / non-existant db name
- targeted db was attached as read_only
- active transaction has already modified db0, modifying db1 in the same transaction shouldn't be possible

Reply:
Thanks for the review, I'll add the tests. Can you clarify a bit on the last test?

Reply:
This isn't a risk uncovered by this new functionality, but it somehow reminded me of this and I'm not sure if we have tests for it.

This would only be a potential problem if the Appender can use an existing transaction, rather than starting and ending its own transaction to execute the appends to the table - which I'm not sure of either

Reply:
basically, pseudo code:

```
start transaction

write to db0.schema.table

do an append to db1.schema.table

commit transaction
```

Hopefully the system errors at either the 3rd step, or otherwise the 4th step, because meta transactions are not allowed to do a write to multiple databases

Reply:
Oh, yea that makes sense - I can add that 👍

Reply:
I'll have to check how the appender does transactions, but I'll look into it.